migrations/002_Beatmaps.up.sql

```diff
@@ -0,0 +1,47 @@
+CREATE TABLE beatmapsets (
+    id serial NOT NULL PRIMARY KEY,
+    title character varying(255) NOT NULL,
+    artist character varying(255) NOT NULL,
+    source character varying(255) NOT NULL,
+    creator_id integer NOT NULL REFERENCES users (id),
+    created_at timestamp without time zone NOT NULL DEFAULT now(),
+    last_updated timestamp without time zone NOT NULL DEFAULT now(),
+    approved_at timestamp without time zone NOT NULL DEFAULT now(),
+    approved_by integer NOT NULL REFERENCES users (id),
+    status smallint NOT NULL DEFAULT -3,
+    description text NOT NULL DEFAULT '',
+    has_video boolean NOT NULL DEFAULT false,
+    availability_status smallint NOT NULL DEFAULT 0,
+    availability_info text NOT NULL DEFAULT ''
+);
+
+CREATE TABLE beatmaps (
+    id serial NOT NULL PRIMARY KEY REFERENCES beatmapsets (id),
```

Comment:
Shouldn't the reference be on `set_id`?
```suggestion
    id serial NOT NULL PRIMARY KEY,
    set_id integer NOT NULL REFERENCES beatmapsets (id),
```

Reply:
No, this targets the beatmapsets table

Reply:
?, why does the beatmap (meaning diff?) id relate to the beatmapset instead of the set id

Reply:
The beatmapset table does not have a `set_id` column. It uses an `id` column.

Reply:
The beatmaps table does have a set_id column though, that relates to nothing

Reply:
Even though I just woke up, I can say that the beatmapsets table does not have that

Reply:
Nevermind, I should never do reviews at 5 am